protocol/communities_events_utils_test.go

```diff
@@ -616,12 +620,62 @@ func kickMember(base CommunityEventsTestsInterface, communityID types.HexBytes,
 
 	s := base.GetSuite()
 	s.Require().NoError(err)
-	s.Require().Nil(checkKicked(response))
 
-	checkClientsReceivedAdminEvent(base, WaitCommunityCondition, checkKicked)
+	// 1. event sender should get pending state for kicked member
+	modifiedCommmunity, err := getModifiedCommunity(response, types.EncodeHex(communityID))
+	s.Require().NoError(err)
+	s.Require().True(modifiedCommmunity.HasMember(&base.GetMember().identity.PublicKey))
+	s.Require().Equal(communities.CommunityMemberKickPending, modifiedCommmunity.PendingAndBannedMembers()[pubkey])
+
+	// 2. wait for event as a sender
+	waitOnMessengerResponse(s, WaitCommunityCondition, func(response *MessengerResponse) error {
+		modifiedCommmunity, err := getModifiedCommunity(response, types.EncodeHex(communityID))
+		if err != nil {
+			return err
+		}
+
+		if !modifiedCommmunity.HasMember(&base.GetMember().identity.PublicKey) {
+			return errors.New("alice should not be not kicked (yet)")
+		}
+
+		if modifiedCommmunity.PendingAndBannedMembers()[pubkey] != communities.CommunityMemberKickPending {
+			return errors.New("alice should be in the pending state")
+		}
+
+		return nil
+	}, base.GetEventSender())
+
+	// 3. wait for event as the community member
+	waitOnMessengerResponse(s, WaitCommunityCondition, func(response *MessengerResponse) error {
+		modifiedCommmunity, err := getModifiedCommunity(response, types.EncodeHex(communityID))
+		if err != nil {
+			return err
+		}
+
+		if !modifiedCommmunity.HasMember(&base.GetMember().identity.PublicKey) {
+			return errors.New("alice should not be not kicked (yet)")
+		}
+
+		if len(modifiedCommmunity.PendingAndBannedMembers()) > 0 {
+			return errors.New("alice should not know about banned and pending members")
+		}
+
+		return nil
+	}, base.GetMember())
+
+	// 4. control node should handle event and actually kick member
+	waitOnMessengerResponse(s, WaitCommunityCondition, checkKicked, base.GetControlNode())
+
+	// 5. event sender get removed member
+	waitOnMessengerResponse(s, WaitCommunityCondition, checkKicked, base.GetEventSender())
```

Comment:
Does this also make sure that there is no longer the kickedPending state?

Reply:
Yes,. checkKicked will fail if we still in pending state